src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -0,0 +1,537 @@
+                "Template for resource group %s has failed validation. The message was: %s.\
+                See logs for additional details.",
+                resource_group,
+                validation_res.error.message,
+            )
+            logger.debug(
+                "Template for resource group %s failed validation. \
+                Full error details: %s",
+                resource_group,
+                validation_res.error,
+            )
+            raise RuntimeError("Azure template validation failed.")
+
+        # Validation succeeded so proceed with deployment
+        logger.debug("Successfully validated resources for %s", resource_group)
+
+        poller = self.api_clients.resource_client.deployments.begin_create_or_update(
+            resource_group_name=resource_group,
+            deployment_name=deployment_name,
+            parameters={
+                "properties": {
+                    "mode": "Incremental",
+                    "template": template,
+                    "parameters": parameters,
+                }
+            },
+        )
+        logger.debug(poller)
+
+        # Wait for the deployment to complete and get the outputs
+        deployment: DeploymentExtended = poller.result()
+        logger.debug("Finished deploying")
+
+        if deployment.properties is not None:
+            depl_props = deployment.properties
+        else:
+            raise RuntimeError("The deployment has no properties.\nAborting")
+        logger.debug("Deployed: %s %s %s", deployment.name, deployment.id, depl_props)
+
+        if depl_props.provisioning_state != "Succeeded":
+            logger.debug("Failed to provision: %s", depl_props)
+            raise RuntimeError(
+                f"Deploy of template to resource group"
+                f" {resource_group} proceeded but the provisioning"
+                f" state returned is {depl_props.provisioning_state}. "
+                f"\nAborting"
+            )
+        logger.debug(
+            "Provisioning state of deployment %s : %s",
+            resource_group,
+            depl_props.provisioning_state,
+        )
+
+        return depl_props.outputs
+
+    def convert_bicep_to_arm(self, bicep_template_path: str) -> Any:
+        """
+        Convert a bicep template into an ARM template.
+
+        :param bicep_template_path: The path to the bicep template to be converted
+                :raise RuntimeError if az CLI is not installed.
+        :return: Output dictionary from the bicep template.
+        """
+        if not shutil.which("az"):
+            logger.error(
+                "The Azure CLI is not installed - follow "
+                "https://github.com/Azure/bicep/blob/main/docs/installing.md#linux"
+            )
+            raise RuntimeError(
+                "The Azure CLI is not installed - cannot render ARM templates."
+            )
+        logger.debug("Converting %s to ARM template", bicep_template_path)
```

Comment:
Is there really a situation where this extension is installed but Azure CLI is not installed?

Reply:
Removed this section of code